Generators/include/Generators/GeneratorTParticleParam.h

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023-2099 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+// @author: Christian Holm Christensen <cholm@nbi.dk>
+#ifndef ALICEO2_EVENTGEN_GENERATORTPARTICLEPARAM_H_
+#define ALICEO2_EVENTGEN_GENERATORTPARTICLEPARAM_H_
+
+#include "CommonUtils/ConfigurableParam.h"
+#include "CommonUtils/ConfigurableParamHelper.h"
+#include <string>
+
+namespace o2
+{
+namespace eventgen
+{
+
+/**
+   a parameter class/struct to keep the settings of the TGenerator
+   event generator and allow the user to modify them */
+struct GeneratorTParticleParam : public o2::conf::ConfigurableParamHelper<GeneratorTParticleParam> {
+  std::string treeName = "T";
+  std::string branchName = "Particles";
+  std::string fileNames = "tparticle.root";
+  std::string progCmd = "";
+  O2ParamDef(GeneratorTParticleParam, "TParticle");
```

Comment:
"TParticle" as key sounds a bit short and might be confusing. Maybe we could go with "TParticleGen" or something (if we decide to keep GeneratorTParticle).

Reply:
I used `TParticle` as the prefix for the keys to stay in line with other prefixes (e.g., `HepMC` for `GeneratorHepMC`, and `Pythia6` for `GeneratorPythia6`).  However, many of the generators key prefix are set to the name of the generator (e.g., `GeneratorPythia8` for `GeneratorPythia8`), so I think it would make more sense to use `GeneratorTParticle` if anything.  On the other hand `TParticleGen` I think is perhaps misleading, as it suggests that there's an actual generator process running, which may or may not be the case. 

Also since the MR, I've done some developments on `GeneratorTParticle` and `GeneratorHepMC` to derive from `GeneratorFileOrCmd` (in addition to deriving from `Generator`).   This base class has utilities for registering input file(s), set up a command line for a child program, FIFOs, temporary file names and so on.   Since the settings for that are common to all generators that derive from that (and `GeneratorFromFile` and `GeneratorO2Kine` could do that), I've introduced the class `GeneratorFileOrCmdParam` with prefix `FileOrCmd`.   That greatly simplifies the code and reduces code duplication quite a bit.   These changes are not yet anywhere to be found, as I'm waiting for this MR to go through.